src/CMakeLists.txt

```diff
@@ -168,4 +167,24 @@ set(tfs_HDR
 	${CMAKE_CURRENT_LIST_DIR}/xtea.h
 	)
 
-add_custom_target(format COMMAND /usr/bin/clang-format -style=file -i ${tfs_HDR} ${tfs_SRC})
+set(tfs_MAIN ${CMAKE_CURRENT_LIST_DIR}/otserv.cpp PARENT_SCOPE)
+
+add_library(tfslib OBJECT ${tfs_SRC})
+target_link_libraries(tfslib PRIVATE
+        Boost::date_time
+        Boost::system
+        Boost::iostreams
+        fmt::fmt
+        ${CMAKE_THREAD_LIBS_INIT}
+        ${Crypto++_LIBRARIES}
+        ${LUA_LIBRARIES}
+        ${MYSQL_CLIENT_LIBS}
+        ${PUGIXML_LIBRARIES}
+        )
+target_include_directories(tfslib PRIVATE ${Boost_INCLUDE_DIRS} ${Crypto++_INCLUDE_DIR} ${LUA_INCLUDE_DIR} ${MYSQL_INCLUDE_DIR} ${PUGIXML_INCLUDE_DIR})
```

Comment:
I had to add the server itself as a library because tests need to be linked against server code. Since I'm using `OBJECT`, it will not generate a library, and the output executable is the same thing as before.